`_ = map[key]` generate the error C codes

```vlang
type Abc = int | string

fn main() {
	x := map[string]Abc{}
	_ = x['nonexisting']
	if y := x['nonexisting'] {
		println(y)
	}
	assert true
}

./t1.v:5:7: warning: `or {}` block required when indexing a map with sum type value
    3 | fn main() {
    4 |     x := map[string]Abc{}
    5 |     _ = x['nonexisting']
      |          ~~~~~~~~~~~~~~~
    6 |     if y := x['nonexisting'] {
    7 |         println(y)
==================
/tmp/v/t1.11856289128462166309.tmp.c:10435: error: ',' expected (got ";")
...
==================
(Use `v -cg` to print the entire error message)
```
generated C code:
```vlang
VV_LOCAL_SYMBOL void main__main(void) {
	Map_string_main__Abc x = new_map(sizeof(string), sizeof(main__Abc), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string);
	{main__Abc _ = map_set(&x, &(string[]){_SLIT("nonexisting")}, &(main__Abc[]) { ;}
	;
	main__Abc* _t2 = (main__Abc*)/*ee elem_ptr_typ */(map_get_check(ADDR(map, x), &(string[]){_SLIT("nonexisting")}));
	Option_main__Abc _t1 = {0};
	if (_t2) {
		*((main__Abc*)&_t1.data) = *((main__Abc*)_t2);
	} else {
		_t1.state = 2; _t1.err = _v_error(_SLIT("array index out of range"));
	}
	
	if (_t1.state == 0) {
		main__Abc y = *(main__Abc*)_t1.data;
		println(main__Abc_str(y));
	}
	// assert
	if (!(true)) {
		VAssertMetaInfo v_assert_meta_info__t3 = {0};
		v_assert_meta_info__t3.fpath = _SLIT("t1.v");
		v_assert_meta_info__t3.line_nr = 8;
		v_assert_meta_info__t3.fn_name = _SLIT("main.main");
		v_assert_meta_info__t3.src = _SLIT("true");
		__print_assert_failure(&v_assert_meta_info__t3);
		_v_panic(_SLIT("Assertion failed..."));
	}
}
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -2925,6 +2925,8 @@ fn (mut g Gen) gen_assign_stmt(assign_stmt ast.AssignStmt) {
 			g.right_is_opt = true
 		}
 		if blank_assign {
+			pre_assign_op := g.assign_op
+			g.assign_op = .decl_assign
 			if is_call {
 				old_is_void_expr_stmt := g.is_void_expr_stmt
 				g.is_void_expr_stmt = true
@@ -2935,6 +2937,7 @@ fn (mut g Gen) gen_assign_stmt(assign_stmt ast.AssignStmt) {
 				g.expr(val)
 				g.writeln(';}')
 			}
+			g.assign_op = pre_assign_op
 			g.is_assign_lhs = false
 		} else if assign_stmt.op == .assign
 			&& (is_fixed_array_init || (right_sym.kind == .array_fixed && val is ast.Ident)) {

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -2925,8 +2925,9 @@ fn (mut g Gen) gen_assign_stmt(assign_stmt ast.AssignStmt) {
 			g.right_is_opt = true
 		}
 		if blank_assign {
-			pre_assign_op := g.assign_op
-			g.assign_op = .decl_assign
+			if val is ast.IndexExpr {
+				g.assign_op = .decl_assign
+			}
 			if is_call {
 				old_is_void_expr_stmt := g.is_void_expr_stmt
 				g.is_void_expr_stmt = true
@@ -2937,7 +2938,6 @@ fn (mut g Gen) gen_assign_stmt(assign_stmt ast.AssignStmt) {
 				g.expr(val)
 				g.writeln(';}')
 			}
-			g.assign_op = pre_assign_op
 			g.is_assign_lhs = false
 		} else if assign_stmt.op == .assign
 			&& (is_fixed_array_init || (right_sym.kind == .array_fixed && val is ast.Ident)) {

--- a/vlib/v/tests/map_get_assign_blank_test.v
+++ b/vlib/v/tests/map_get_assign_blank_test.v
@@ -1,8 +1,17 @@
 type Abc = int | string
 
-fn test_map_get_assign_blank() {
+fn test_map_get_decl_assign_blank() {
 	x := map[string]Abc{}
 	_ := x['nonexisting']
+	if y := x['nonexisting'] {
+		println(y)
+	}
+	assert true
+}
+
+fn test_map_get_assign_blank() {
+	x := map[string]Abc{}
+	_ = x['nonexisting']
 	if y := x['nonexisting'] {
 		println(y)
 	}

